packages/waku/src/config.ts

```diff
@@ -49,14 +49,16 @@ export interface Config {
    */
   middleware?: () => Promise<{ default: Middleware }>[];
   /**
-   * Enhander for Hono
+   * Enhancer for Hono
    * Defaults to `undefined`
    */
-  unstable_honoEnhancer?:
-    | (<Hono>(createApp: (app: Hono) => Hono) => (app: Hono) => Hono)
-    | undefined;
+  unstable_honoEnhancer?: unstable_HonoEnhancer | undefined;
 }
 
+export type unstable_HonoEnhancer<Hono = never, MiddlewareHandler = never> = (
+  createApp: (app: Hono) => Hono,
+) => Promise<(app: Hono, serveWaku: MiddlewareHandler) => Hono>;
```

Comment:
`Promise<>` is fine, but exposing `serveWaku` is too much.

Reply:
Another idea would be:
```ts
(createApp: (app: Hono) => Promise<Hono>) => (app: Hono) => Promise<Hono>
```

Reply:
> accept an async function to support dynamic imports.

Hm, on second thought, I wonder if promise is really necessary. Do you really need dynamic imports?

Can you try top-level await?

At this point, it's not convincing to change `unstable_honoEnhancer`.

Reply:
> Can you try top-level await?

Yes, I can try. It needs to import for the `waku dev` command and not import during build. I'm not sure if there is a better way to do that than using `import.meta.env.PROD`.

>  exposing serveWaku is too much. It doesn't look like a nice enhancer abstraction.

Exposing the built Hono middleware (serveWaku) is useful if the user doesn't want Waku to be the default route. I agree that it doesn't look as clean. Maybe we should leave it out until someone has a more compelling use case? I am fine with transforming the build output for my personal use cases.

I will make adjustments to this PR to remove the API changes.

I will also try to fix the error it throws when deployed to Cloudflare. I tried using top-level await there but the same error occurred.

Reply:
> useful if the user doesn't want Waku to be the default route.

We expose `runner` for such use cases. I don't think they would use `unstable_honoEnhancer` in such cases. But, maybe I miss some use cases.

> I agree that it doesn't look as clean.

I would like it to be clean. 😄

Reply:
> We expose `runner` for such use cases. I don't think they would use `unstable_honoEnhancer` in such cases. But, maybe I miss some use cases.

OK great. I will try that for the use case I am thinking of.

Reply:
> > accept an async function to support dynamic imports.
> 
> Hm, on second thought, I wonder if promise is really necessary. Do you really need dynamic imports?
> 
> Can you try top-level await?

The build fails if wrangler and miniflare are imported. I'm pretty sure there isn't another way around that than conditionally importing with a dynamic import, but I might be missing something.

Reply:
Yeah... I think there probably is a technique with initializing promises and then awaiting them inside the route handler - similar to how serverEngine is created at the top level and awaited in the handler in some of the platform server code. Thanks. I'll keep working on it.

Reply:
actually the production mode check can be done outside `app.use` in this case? anyway, hope you can work it around.

Reply:
I got this working nicely. https://gist.github.com/rmarscher/9bb6ed54dc9535f4b81bed147204c7e9